File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    public List<TbBrand> findAll();

    /**
     * 品牌分页
     *
     * @param pageNum  当前页码
     * @param pageSize 每页显示条数
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);

    /**
     * 新增品牌
     *
     * @param tbBrand
     */
    public void add(TbBrand tbBrand);

    /**
     * 更加id查找
     */
    public TbBrand findOne(Long id);

    /**
     * 修改品牌
     */
    public void update(TbBrand tbBrand);

    public void delete(Long[] ids);

    /**
     * 模糊分页查询
     */
    public PageResult findPage(TbBrand tbBrand, int pageNum, int pageSize);

    /**
     * 返回下拉列表
     * @return
     */
    public List<Map> selectOptionList();


}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务   先有服务后有控制层，服务请求后端后返回数据（model)，控制层操作demol，model最后渲染视图view
app.service('brandService', function ($http) {
    this.findAll = function () {
        return $http.get('../brand/findAll.do');
    };

    this.findPage = function (page, size) {
        return $http.get('../brand/findPage.do?page=' + page + '&size=' + size);
    };

    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id=' + id);
    };

    this.add = function (entity) {
        return $http.post('../brand/add.do', entity)
    };

    this.update = function (entity) {
        return $http.post('../brand/update.do', entity)
    };
    this.delete = function (ids) {
        return $http.get('../brand/delete.do?ids=' + ids);

    };
    this.search = function (page, size, searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&size=' + size, searchEntity);
    }
    this.selectOptionList=function () {
        return $http.get('../brand/selectOptionList.do')
    }
});